File: ios/JitsiMeetViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(JitsiMeetViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(options, NSDictionary)
  RCT_EXPORT_VIEW_PROPERTY(onConferenceTerminated, RCTDirectEventBlock)
  RCT_EXPORT_VIEW_PROPERTY(onConferenceJoined, RCTDirectEventBlock)
  RCT_EXPORT_VIEW_PROPERTY(onConferenceWillJoin, RCTDirectEventBlock)
@end

File: ios/JitsiMeetViewManager.swift
import UIKit
import React
import JitsiMeetSDK

@objc(JitsiMeetViewManager)
class JitsiMeetViewManager: RCTViewManager {
  override func view() -> UIView! {
    return RNJitsiMeetView()
  }
  
  override class func requiresMainQueueSetup() -> Bool {
    return true
  }
}
